src/pools/D3MAavePool.sol

```diff
@@ -165,7 +166,11 @@ contract D3MAavePool is ID3MPool {
     // Withdraws Dai from Aave in exchange for adai
     // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#withdraw
     function withdraw(uint256 wad) external override onlyHub {
-        pool.withdraw(address(dai), wad, address(msg.sender));
+        uint256 prevDai = dai.balanceOf(msg.sender);
```

Comment:
See section 7.1 in the ChainSecurity audit -
"Upon withdrawal the D3MHub enforces that a sufficient amount of DAI is present. Otherwise, the transaction reverts when DAIJOIN.join() fails.
Generally, this checks whether redeeming pool shares resulted in the expected amount of DAI, except in corner cases when there was additional DAI balance present for the D3MHub."

Reply:
Ah yeah I remember that, still not sure if the check is worth it. I mean in normal operations dai shouldn't be in the hub, and if it is is because someone donated it anyway.